crates/nu-protocol/src/signature.rs

```diff
@@ -143,14 +132,35 @@ impl Signature {
             required_positional: vec![],
             optional_positional: vec![],
             rest_positional: None,
-            named: vec![flag],
+            named: vec![],
             is_filter: false,
             creates_scope: false,
             category: Category::Default,
         }
     }
 
+    // Add a default help option to a signature
+    pub fn add_help(mut self) -> Signature {
+        // default help flag
+        let flag = Flag {
+            long: "help".into(),
+            short: Some('h'),
+            arg: None,
+            desc: "Display this help message".into(),
+            required: false,
+            var_id: None,
+            default_value: None,
+        };
+        self.named.push(flag);
+        self
+    }
+
+    // Build an internal signature with default help option
     pub fn build(name: impl Into<String>) -> Signature {
+        Signature::new(name.into()).add_help()
+    }
+    // Build an external signature with no presumed flags.
+    pub fn build_external(name: impl Into<String>) -> Signature {
```

Comment:
This is not called anywhere. Is your fix behaving as you intended or an unwanted side effect?

Reply:
I think it is behaving as intended. I added build_external so that we could use the same pattern to create external definitions as we do internal ones, without changing every existing call to Signature::build. The more substantive change is the fact that a Signautre::new() no longer adds the help flag by default. The parser calls a new method, add_help, for definitions, and not for externals.

I'm not sure it would ever make sense to define an 'extern' command programmatically outside of a test. And I didn't even end up using it in my testing, so I've dropped it.

Reply:
Yeah, I am not hundred percent sure how to future proof it but you documented the behavior properly so let's land that now.